src/Test/FunctorTest.hs

```diff
@@ -72,6 +73,14 @@ anonMapTest =
       x <$ Full (q :: Integer) == Full (x :: Integer)
   ]
 
+functorFunTest :: TestTree
+functorFunTest =
+  testGroup "??" [
+    testCase "Map with List" $ (((*2) :. (+1) :. const 99 :. Nil) ?? 8) @?= (16 :. 9 :. 99 :. Nil)
+  , testCase "Map with Optional" $ (Full (+1) ?? 8) @?= Full 9
+  , testCase "Map with Optional Empty" $ ((Empty :: Num a => Optional (a -> a)) ?? 8) @?= Empty
```

Comment:
Could we do `Empty :: Optional (Int -> Int)`? I think that would be one less thing for a reader to think about.

Reply:
comment addressed